README.md

```diff
@@ -65,30 +65,39 @@ For more information please refer to [Installation Guide](./INSTALLATION.md).
 
 ## Contributing
 
+Please read [CONTRIBUTING](./CONTRIBUTING.md) if you are new here or not familiar with the basic rules of Git/GitHub world.
+
 ### Quickstart
 
-```bash
-git clone https://github.com/hypertrons/hypertrons-crx
-cd hypertrons-crx
-npm install
-npm run watch
-npm run web-ext
-```
-
-[web-ext](https://github.com/mozilla/web-ext) would open `Chrome` and load `Hypertrons-crx` into the browser automatically. And the default configuration of `web-ext` can be found in [package.json](https://github.com/hypertrons/hypertrons-crx/blob/master/package.json):
-
-```json
-{
-  "webExt": {
-    "sourceDir": "distribution",
-    "run": {
-      "keepProfileChanges": true,
-      "chromiumProfile": "./test/web-ext-profile",
-      "startUrl": ["https://github.com/hypertrons/hypertrons-crx"]
-    }
-  }
-}
-```
+1. git clone https://github.com/hypertrons/hypertrons-crx
+
+2. cd hypertrons-crx
+
+3. yarn install
+
+4. yarn run start
+
+5. Load the freshly built unpacked extension on Chrome following:
+
+   1. Access chrome://extensions/
+
+   2. Check "Developer mode"
+
+   3. Click on "Load unpacked extension"
+
+   4. Select the "build" folder under the project root directory
+
+   5. Keep "service worker" DevTools page open ([why?]())
+
+      ![](./assets/keep-service-worker-devtools-open.jpeg)
+
+6. Happy hacking!
+
+### HMR & auto-reload
+
+If you are developing Options page or Popup page, each time you save files the pages will hot replace the modules without refreshing, which means you can see the changes right away.
+
+However, if you are developing Background or ContentScripts, each time you save files the service worker will reload the extension automatically. And if you are developing ContentScripts, then pages that injected with ContentScripts will refresh themselves to run the newest scripts.
```

Comment:
In Chrome extension development, some entries does not support [HMR (Hot Module Replacement)](https://webpack.js.org/concepts/hot-module-replacement/). Two typical examples are the Content Scripts (which lack the permission to establish a connection with the Dev Server to receive new changes) and the Background script (which is now a service worker under manifest v3 and has no global document node that the HMR requires to work).

One possible solution to achieve a similar effect is to inject clients in the Background script and the Content Scripts and set-up a customized middleware in the Dev Server. The client in the Background script establishes a long connection with the Dev Server. When the Dev Server announces a code change in the Background scripts, the client in the Background scripts reload the entire extension with chrome.runtime.reload(); when the Dev Server announces a code change in the Content Scripts, the client in the Background script communicate with the client in the Content Scripts with chrome.runtime.sendMessage(), and the client in the Content Scripts refresh the page. This Auto-Reload and Auto-Refresh approach makes the new changes in the code to be applied immediately and eliminates some tiny troubles. However, unlike HMR, which can preserve state of the code, this approach will inevitably lose all state of the code, which can be unwanted in some cases.

Notice that the background service worker will be inactive automatically after a period of time, which will cause both the Auto-Reload and Auto-Refresh functionality to be non-functional. One possible solution to fix this issue is to keep the DevTool panel of the Background script open during the development